ecosystem/typescript/sdk_v2/src/crypto/multi_ed25519.ts

```diff
@@ -0,0 +1,193 @@
+    this.publicKeys.forEach((k: PublicKey, i: number) => {
+      bytes.set(k.toUint8Array(), i * PublicKey.LENGTH);
+    });
+
+    bytes[this.publicKeys.length * PublicKey.LENGTH] = this.threshold;
+
+    return bytes;
+  }
+
+  serialize(serializer: Serializer): void {
+    serializer.serializeBytes(this.toUint8Array());
+  }
+
+  static deserialize(deserializer: Deserializer): MultiPublicKey {
+    const bytes = deserializer.deserializeBytes();
+    const threshold = bytes[bytes.length - 1];
+
+    const keys: PublicKey[] = [];
+
+    for (let i = 0; i < bytes.length - 1; i += PublicKey.LENGTH) {
+      const begin = i;
+      keys.push(new PublicKey({ hexInput: bytes.subarray(begin, begin + PublicKey.LENGTH) }));
+    }
+    return new MultiPublicKey({ publicKeys: keys, threshold });
+  }
+}
+
+export class MultiSignature {
+  // Maximum number of signatures supported
+  static MAX_SIGNATURES_SUPPORTED = 32;
+
+  // Bitmap length
+  static BITMAP_LEN: number = 4;
+
+  public readonly signatures: Signature[];
+
+  public readonly bitmap: Uint8Array;
+
+  /**
+   * Signature for a K-of-N multisig transaction.
+   *
+   * @see {@link
+   * https://aptos.dev/integration/creating-a-signed-transaction/#multisignature-transactions | Creating a Signed Transaction}
+   *
+   * @param args.signatures A list of signatures
+   * @param args.bitmap 4 bytes, at most 32 signatures are supported. If Nth bit value is `1`, the Nth
+   * signature should be provided in `signatures`. Bits are read from left to right
+   */
+  constructor(args: { signatures: Signature[]; bitmap: Uint8Array }) {
+    const { signatures, bitmap } = args;
+    if (bitmap.length !== MultiSignature.BITMAP_LEN) {
+      throw new Error(`"bitmap" length should be ${MultiSignature.BITMAP_LEN}`);
+    }
+
+    this.signatures = signatures;
+    this.bitmap = bitmap;
+  }
+
+  /**
+   * Converts a MultiSignature into Uint8Array (bytes) with `bytes = s1_bytes | ... | sn_bytes | bitmap`
+   */
+  toUint8Array(): Uint8Array {
+    const bytes = new Uint8Array(this.signatures.length * Signature.LENGTH + MultiSignature.BITMAP_LEN);
+    this.signatures.forEach((k: Signature, i: number) => {
+      bytes.set(k.toUint8Array(), i * Signature.LENGTH);
+    });
+
+    bytes.set(this.bitmap, this.signatures.length * Signature.LENGTH);
+
+    return bytes;
+  }
+
+  /**
+   * Helper method to create a bitmap out of the specified bit positions
+   * @param bits The bitmap positions that should be set. A position starts at index 0.
+   * Valid position should range between 0 and 31.
+   * @example
+   * Here's an example of valid `bits`
+   * ```
+   * [0, 2, 31]
+   * ```
+   * `[0, 2, 31]` means the 1st, 3rd and 32nd bits should be set in the bitmap.
+   * The result bitmap should be 0b1010000000000000000000000000001
+   *
+   * @returns bitmap that is 32bit long
+   */
+  static createBitmap(bits: number[]): Uint8Array {
```

Comment:
should it be
```suggestion
  static createBitmap(args:{bits: number[]}): Uint8Array {
```

Reply:
Curious, do we want to make everything obj for argument, including Serialize and Deserialize method as well?